### PR TITLE
Cabal.project <-> stack.yaml correspondence

### PR DESCRIPTION
**Pull request by deepfire**
`cabal.project` and `stack.yaml` keep going out of sync.

This PR does:

1. restore their correspondence
2. reformats them, so that it's easier to maintain the correspondence manually.

The second part can be controversial, so let's discuss!

### CODE REVIEW

File: stack.yaml [hunk 1]
```diff
@@ -9,7 +9,30 @@ ghc-options:
...
+
+  - git: https://github.com/input-output-hk/cardano-base
+    commit: 0e26a4f00324100b4c34f2a9c7606ae03e94d7a2
+    subdirs:
+      - binary
+      - binary/test
+      - cardano-crypto-class
+
+
+
```
Comment (erikd) [inline stack.yaml]: What is all this white space?

File: stack.yaml [hunk 2]
```diff
@@ -9,7 +9,30 @@ ghc-options:
...
+
+
+
+
+
+  # - git: https://github.com/input-output-hk/cardano-crypto
+  #   commit: 4590efa638397e952a51a8994b5543e4ea3c1ecd
```
Comment (erikd) [inline stack.yaml]: If that is not needed, it should be removed.

File: cabal.project [hunk 3]
```diff
@@ -124,6 +125,11 @@ source-repository-package
   tag: 2ad7dced59caece77b6a5e06914aa37d8a3bba23
   subdir: io-sim-classes
 
+source-repository-package
+  type: git
+  location: https://github.com/joelwilliamson/bimap
+  tag: 013a0639c222b3f5572ae20ce364815e5d8f2156
+
```
Comment (erikd) [inline cabal.project]: We should not be using the git version here. `bimap-0.4` should be sufficient.

**Comment by deepfire** [comment]
Deemed unnecessary.

**Pull request closed without merging**